test/skimNtuple_HHbtag.cpp

```diff
@@ -3361,102 +3422,108 @@ int main (int argc, char** argv)
-				  double Eff_MC_ele_down   = passSingle * SFL_MC_down   - passCross * passSingle * std::min(SFl_MC_down  , SFL_MC_down)   * SFtau_MC   + passCross * SFl_MC_down   * SFtau_MC;
-
-				  // for each DM, fill a trigSF branch with the up/down values if the tauh has the corresponding DM, otherwise fill with nominal trigSF value
-				  vector <double> SFtau_Data_up   (N_tauhDM, SFtau_Data);
-				  vector <double> SFtau_Data_down (N_tauhDM, SFtau_Data);
-				  vector <double> SFtau_MC_up     (N_tauhDM, SFtau_MC);
-				  vector <double> SFtau_MC_down   (N_tauhDM, SFtau_MC);
-				  vector <double> Eff_Data_up     (N_tauhDM, Eff_Data);
-				  vector <double> Eff_Data_down   (N_tauhDM, Eff_Data);
-				  vector <double> Eff_MC_up       (N_tauhDM, Eff_MC);
-				  vector <double> Eff_MC_down     (N_tauhDM, Eff_MC);
-
-				  for (int idm  = 0; idm < N_tauhDM; idm ++)
-					{
-					  if (isthisDM_second[idm])
-						{
-						  SFtau_Data_up[idm]   = tauTrgSF_etau->getEfficiencyData(tlv_secondLepton.Pt(), DM2, 1);
-						  SFtau_Data_down[idm] = tauTrgSF_etau->getEfficiencyData(tlv_secondLepton.Pt(), DM2, -1);
-						  SFtau_MC_up[idm]     = tauTrgSF_etau->getEfficiencyMC  (tlv_secondLepton.Pt(), DM2, 1);
-						  SFtau_MC_down[idm]   = tauTrgSF_etau->getEfficiencyMC  (tlv_secondLepton.Pt(), DM2, -1);
+				  double Eff_Data = passSingle * Eff_SL_ele_Data - passCross * passSingle * std::min(Eff_cross_ele_Data, Eff_SL_ele_Data) * Eff_cross_tau_Data + passCross * Eff_cross_ele_Data * Eff_cross_tau_Data;
+				  double Eff_MC   = passSingle * Eff_SL_ele_MC   - passCross * passSingle * std::min(Eff_cross_ele_MC  , Eff_SL_ele_MC)   * Eff_cross_tau_MC   + passCross * Eff_cross_ele_MC   * Eff_cross_tau_MC;
+
+				  // Compute uncertainties (separate for SingleElectron and ele leg of cross trigger)
+				  double Err_Data_SL_ele = passSingle * Eff_SL_ele_Data_Err - passCross * passSingle * (Eff_cross_ele_Data > Eff_SL_ele_Data) * Eff_SL_ele_Data_Err * Eff_cross_tau_Data;
+				  double Err_MC_SL_ele   = passSingle * Eff_SL_ele_MC_Err   - passCross * passSingle * (Eff_cross_ele_MC   > Eff_SL_ele_MC)   * Eff_SL_ele_MC_Err   * Eff_cross_tau_MC;
+
+				  double trigSF_SL_ele_err = 0.;
+				  if(passSingle) {
+					trigSF_SL_ele_err = eTrgSF->get_ScaleFactorError(Eff_Data, Eff_MC, Err_Data_SL_ele, Err_MC_SL_ele);
+				  }
+
+				  double Err_Data_cross_ele = - passCross * passSingle * (Eff_cross_ele_Data <= Eff_SL_ele_Data) * Eff_cross_ele_Data_Err * Eff_cross_tau_Data + passCross * Eff_cross_ele_Data_Err * Eff_cross_tau_Data;
+				  double Err_MC_cross_ele   = - passCross * passSingle * (Eff_cross_ele_MC   <= Eff_SL_ele_MC)   * Eff_cross_ele_MC_Err   * Eff_cross_tau_MC   + passCross * Eff_cross_ele_MC_Err   * Eff_cross_tau_MC;
+
+				  double trigSF_cross_ele_err = 0.;
+
+				  // if single lepton trigger and cross trigger are passed and the cross trigger is less efficient than
```

Comment:
you seem to be missing a negation before "single lepton trigger" (looking at the ```if``` condition)

but I admit I'm still very much confused by this check

Reply:
Tobias kindly explained me the logic privately. All good.

Reply:
Hopefully also the comments somewhat explains it now